File: theMealDB.py
import requests


# last numer is API key, 1 is public testing key, can be used for low amount of requests but might be removed
API = 'http://www.themealdb.com/api/json/v1/1/'

# load random meal
RANDOM = 'random.php'

# load details based on meal id
DETAILS = 'lookup.php?i='

# load latest meals
LATEST = 'latest.php'

# list meal categories
CATEGORIES = 'list.php?c=list'

# list meal areas
AREAS = 'list.php?a=list'

# list ingredients
INGREDIENTS = 'list.php?i=list'

# filter meals by category
FILTER_CATEGORY = 'filter.php?c='

# filter meals by area
FILTER_AREA = 'filter.php?a='

# filter meals by ingredient
FILTER_INGREDIENT = 'filter.php?i='


def get_random():
    url = API + RANDOM
    return requests.get(url)


def get_details(meal_id):
    url = API + DETAILS + meal_id
    return requests.get(url)


def get_latest():
    url = API + LATEST
    return requests.get(url)


def get_categories():
    url = API + CATEGORIES
    return requests.get(url)


def get_areas():
    url = API + AREAS
    return requests.get(url)


def get_ingredients():
    url = API + INGREDIENTS
    return requests.get(url)


def filter_category(category):
    url = API + FILTER_CATEGORY + category
    return requests.get(url)


def filter_area(area):
    url = API + FILTER_CATEGORY + area
    return requests.get(url)


def filter_ingredient(ingredient):
    url = API + FILTER_CATEGORY + ingredient
    return requests.get(url)

File: theMealDB.mjs
/** Fetch meal data from the meal db (https://www.themealdb.com/)
 *  information about API in forum (https://forum.kodi.tv/showthread.php?tid=282387)
 * **/

export default class TheMealDB {

    constructor(props) {
    }


    // last numer is API key, 1 is public testing key, can be used for low amount of requests but might be removed
    static API = 'https://www.themealdb.com/api/json/v1/1/';

    // load random meal
    static RANDOM = 'random.php';

    // load details based on meal id
    static DETAILS = 'lookup.php?i=';

    // load latest meals
    static LATEST = 'latest.php';

    // list meal categories
    static CATEGORIES = 'list.php?c=list';

    // list meal areas
    static AREAS = 'list.php?a=list';

    // list ingredients
    static INGREDIENTS = 'list.php?i=list';

    // filter meals by category
    static FILTER_CATEGORY = 'filter.php?c=';

    // filter meals by area
    static FILTER_AREA = 'filter.php?a=';

    // filter meals by ingredient
    static FILTER_INGREDIENT = 'filter.php?i=';


    static get_random() {
        const url = TheMealDB.API + TheMealDB.RANDOM;
        const xmlHttp = new XMLHttpRequest();
        xmlHttp.open('GET', url, false);
        xmlHttp.send();
        const response = xmlHttp.response;
        return JSON.parse(response);
    }


    static get_details(meal_id) {
        const url = TheMealDB.API + TheMealDB.DETAILS + meal_id;
        const xmlHttp = new XMLHttpRequest();
        xmlHttp.open('GET', url, false);
        xmlHttp.send();
        return xmlHttp.responseText;
    }

    static get_latest() {
        const url = TheMealDB.API + TheMealDB.LATEST;
        const xmlHttp = new XMLHttpRequest();
        xmlHttp.open('GET', url, false);
        xmlHttp.send();
        return xmlHttp.responseText;
    }


    static get_categories() {
        const url = TheMealDB.API + TheMealDB.CATEGORIES;
        const xmlHttp = new XMLHttpRequest();
        xmlHttp.open('GET', url, false);
        xmlHttp.send();
        return xmlHttp.responseText;
    }


    static get_areas() {
        const url = TheMealDB.API + TheMealDB.AREAS;
        const xmlHttp = new XMLHttpRequest();
        xmlHttp.open('GET', url, false);
        xmlHttp.send();
        return xmlHttp.responseText;
    }


    static get_ingredients() {
        const url = TheMealDB.API + TheMealDB.INGREDIENTS;
        const xmlHttp = new XMLHttpRequest();
        xmlHttp.open('GET', url, false);
        xmlHttp.send();
        return xmlHttp.responseText;
    }


    static filter_category(category) {
        const url = TheMealDB.API + TheMealDB.FILTER_CATEGORY + category;
        const xmlHttp = new XMLHttpRequest();
        xmlHttp.open('GET', url, false);
        xmlHttp.send();
        return xmlHttp.responseText;
    }


    static filter_area(area) {
        const url = TheMealDB.API + TheMealDB.FILTER_AREA + area;
        const xmlHttp = new XMLHttpRequest();
        xmlHttp.open('GET', url, false);
        xmlHttp.send();
        return xmlHttp.responseText;
    }


    static filter_ingredient(ingredient) {
        const url = TheMealDB.API + TheMealDB.FILTER_INGREDIENT + ingredient;
        const xmlHttp = new XMLHttpRequest();
        xmlHttp.open('GET', url, false);
        xmlHttp.send();
        return xmlHttp.responseText;
    }

}